src/robotide/editor/kweditor.py

```diff
@@ -568,24 +567,23 @@ def OnKeyDown(self, event):
             if keycode == wx.WXK_SPACE:
                 self._open_cell_editor_with_content_assist()  # Mac CMD
             elif specialkcode in [wx.WXK_DOWN, wx.WXK_UP]:
-                self._skip_except_on_mac(event)
                 self._move_rows(specialkcode)
             elif specialkcode == wx.WXK_RETURN:
-                self._move_cursor_down(event)
-            else:
-                event.Skip()
+                event.GetEventObject().WriteText('\n')
+                return
         else:
             if specialkcode == wx.WXK_WINDOWS_MENU:
                 self.OnCellRightClick(event)
-            elif specialkcode in [wx.WXK_RETURN, wx.WXK_BACK]:
-                self.save()
-                self._move_grid_cursor(event, specialkcode)
+            elif specialkcode == wx.WXK_BACK:
+                self.OnDelete(event)
+                return
+            elif specialkcode == wx.WXK_RETURN:
+                if not self.IsCellEditControlShown():
+                    self._open_cell_editor()
+                    return
```

Comment:
I know you're trying to make it work like in a spreadsheet, but I think this way is more intuitive when editing RF tests.

For enter button:
- when not in edit mode, will enter edit mode of current cell
- when in edit mode, will exit edit mode and select cell to its right

For back button:
- select cell to the left
This is primarily used to go back to the cell on the left, after enter button is used. Deleting the content would be confusing for those used to work this way.


```
            elif specialkcode == wx.WXK_BACK:
                self._move_grid_cursor(event, specialkcode)
            elif specialkcode == wx.WXK_RETURN:
                if self.IsCellEditControlShown():
                    self._move_grid_cursor(event, specialkcode)
                else:
                    self._open_cell_editor()
```

Reply:
Nyral explained perfectly the current and desired behavior.

Reply:
- for the back button, I mix up the back & delete button, I'll revert it.
- for the enter button, @Nyral you suggestion is a better solution.
```
                if self.IsCellEditControlShown():
                    self._move_grid_cursor(event, specialkcode)
                else:
                    self._open_cell_editor()
```